### DISCUSSION TITLE
Open kitty and run a command

### DISCUSSION THREAD

**Discussion by dalvand** [kovidgoyal/kitty#7197] [General]
How to open another kitty os-window and run a command like `ls` in it all in one command?

**Comment by kovidgoyal** [comment]
https://sw.kovidgoyal.net/kitty/launch/

**Comment by dalvand** [comment]
@kovidgoyal how can I use launch command in a terminal (other than kitty) to achieve what I asked in my post? I want a single command e.g. 
$ kitty ....
Thank you

**Reply by kovidgoyal** [reply]
kitty --hold ls

**Reply by kovidgoyal** [reply]
And if you want to run it in the same instance then enable remote control and use

kitten @ launch --hold ls

**Reply by dalvand** [reply]
@kovidgoyal thank you very much

**Comment by Luflosi** [comment]
See https://sw.kovidgoyal.net/kitty/remote-control/ if you want to control kitty from another terminal.

**Comment by dalvand** [comment]
@Luflosi thanks, but I am trying to open a new kitty os-window that runs a command like `ls` at startup all in one command.

**Comment by Luflosi** [comment]
Start kitty with `kitty -o allow_remote_control=yes --listen-on unix:/tmp/mykitty` to allow remote control from the other terminal, then open a new OS window and execute `ls` in the shell:
```sh
kitten @ --to unix:/tmp/mykitty launch --type os-window sh -c 'bash --rcfile <(echo ". ~/.bashrc; ls")'
```

**Comment by dalvand** [comment]
@Luflosi This doesn't work. The first command opens a new os-window (window A). But the second command doesn't run ls in window A. Any suggestion?
